pkg/conversation/conversation.go

```diff
@@ -0,0 +1,356 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversation
+
+import (
+	piiscrubber "github.com/aavaz-ai/pii-scrubber"
+	contribConverse "github.com/dapr/components-contrib/conversation"
+	"github.com/dapr/dapr/pkg/messages"
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/kit/ptr"
+)
+
+// ConvertUsageToProto converts components-contrib UsageInfo to Dapr protobuf ConversationUsage
+func ConvertUsageToProto(resp *contribConverse.ConversationResponse) *runtimev1pb.ConversationUsage {
+	if resp == nil || resp.Usage == nil {
+		return nil
+	}
+
+	// Safely convert uint64 to uint32 for token counts
+	// Token counts realistically never exceed uint32 range (~4.3 billion)
+	return &runtimev1pb.ConversationUsage{
+		PromptTokens:     ptr.Of(uint32(resp.Usage.PromptTokens)),
+		CompletionTokens: ptr.Of(uint32(resp.Usage.CompletionTokens)),
+		TotalTokens:      ptr.Of(uint32(resp.Usage.TotalTokens)),
+	}
+}
+
+// NeedsInputScrubber checks if PII scrubbing is needed for any input
+func NeedsInputScrubber(req *runtimev1pb.ConversationRequest) bool {
+	for _, input := range req.GetInputs() {
+		if input.GetScrubPII() {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Surely we should only be doing the scrubbing on inputs where it is enabled? 
This is returning true for all (?).